pkg/nsm/interfacename/server.go

```diff
@@ -44,14 +43,14 @@ func NewServer(prefix string, generator NameGenerator) networkservice.NetworkSer
 // A non-nil error is returned if the name generation fails, if a next element in the chain returns a non-nil error
 // It implements NetworkServiceServer for the interfacename package
 func (ine *interfaceNameServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
-	// TODO: check if interface name already exists
-	ine.SetInterfaceName(request, metadata.IsClient(ine))
+	ine.SetInterfaceName(request)
 	return next.Server(ctx).Request(ctx, request)
 }
 
 // Close it does nothing except calling the next Close in the chain
 // A non-nil error if a next element in the chain returns a non-nil error
 // It implements NetworkServiceServer for the interfacename package
 func (ine *interfaceNameServer) Close(ctx context.Context, conn *networkservice.Connection) (*empty.Empty, error) {
+	ine.UnsetInterfaceName(conn)
```

Comment:
I think this mostly fits the TAPA use case. Although if the connect did not succeed at all, and the tapa user "disconnects" the trench-conduit-stream and does not intend to connect it again (i.e. connects a different conduit instead), then the interface name won't get released.
In general probably release on failed request would be more "fault proof", than caching. But I leave it up to you to decide.

Reply:
I am not sure if releasing on a request fail could work since the request can be used to update a connection. It is fails on an update, then the name will be released will it is still in use.

Reply:
Right. Then probably some states would need to be associated with the interface names indicating if they belonged to established connections or not. And names of established connections would not be allowed to get removed upon failed Requests.
But I don't think we should spend more time on this one.